Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7011,6 +7154,1999 @@ def panorama_install_file_content_update_command(args: dict):
+        )
+        params.append(
+            VersionedParamPath("is_allow", vartype="exist", path="action/allow")
+        )
+        params.append(
+            VersionedParamPath("is_drop", vartype="exist", path="action/drop")
+        )
+        params.append(
+            VersionedParamPath("is_block_ip", vartype="exist", path="action/block-ip")
+        )
+        self._params = tuple(params)
+
+
+class AntiSpywareProfile(CustomVersionedPanObject):
+    """Vulnerability Profile Group Object
+
+    Args:
+        name (str): Name of the object
+
+    """
+
+    ROOT = Root.VSYS  # pylint: disable=E1101
+    SUFFIX = ENTRY
+    CHILDTYPES = (AntiSpywareProfileRule,)
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value="/profiles/spyware")
+
+        # params
+        params = []
+
+        self._params = tuple(params)
+
+
+class VulnerabilityProfileRule(VersionedPanObject):
+    """Vulnerability Profile Rule Object
+
+    Args:
+        name (str): Name of the object
+        severity (list:str): List of severities matching this rule
+    """
+    ROOT = Root.VSYS  # pylint: disable=E1101
+    SUFFIX = ENTRY
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value="/rules")
+
+        # params
+        params = []
+        params.append(
+            VersionedParamPath("severity", vartype="member", path="severity")
+        )
+        params.append(
+            VersionedParamPath("is_reset_both", vartype="exist", path="action/reset-both")
+        )
+        params.append(
+            VersionedParamPath("is_reset_client", vartype="exist", path="action/reset-client")
+        )
+        params.append(
+            VersionedParamPath("is_reset_server", vartype="exist", path="action/reset-server")
+        )
+        params.append(
+            VersionedParamPath("is_alert", vartype="exist", path="action/alert")
+        )
+        params.append(
+            VersionedParamPath("is_default", vartype="exist", path="action/default")
+        )
+        params.append(
+            VersionedParamPath("is_allow", vartype="exist", path="action/allow")
+        )
+        params.append(
+            VersionedParamPath("is_drop", vartype="exist", path="action/drop")
+        )
+        params.append(
+            VersionedParamPath("is_block_ip", vartype="exist", path="action/block-ip")
+        )
+        self._params = tuple(params)
+
+
+class VulnerabilityProfile(CustomVersionedPanObject):
+    """Vulnerability Profile Group Object
+
+    Args:
+        name (str): Name of the object
+    """
+
+    ROOT = Root.VSYS  # pylint: disable=E1101
+    SUFFIX = ENTRY
+    CHILDTYPES = (VulnerabilityProfileRule,)
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value="/profiles/vulnerability")
+
+        # params
+        params = []
+
+        self._params = tuple(params)
+
+
+class URLFilteringProfile(VersionedPanObject):
```

Comment:
please elaborate what is the purpose of these classes inheriting from VersionedPanObject

Reply:
These classes are used directly by the pan-os-python package. In some cases, pan-os-python does not implement an object model for the type of configuration we need to look for. Instead of falling back to the raw XML api,  we can simply write the models ourselves, and as long as we inherit from VersionedPanObject, these can be used in the exact same way any other object can.

In the long term we would contribute these models back to the pan-os-python project so we could pull them out of this code.